lib/mayaUsd/ufe/UsdHierarchyHandler.cpp

```diff
@@ -41,6 +41,9 @@ UsdHierarchyHandler::Ptr UsdHierarchyHandler::create()
 Ufe::Hierarchy::Ptr UsdHierarchyHandler::hierarchy(const Ufe::SceneItem::Ptr& item) const
 {
     UsdSceneItem::Ptr usdItem = std::dynamic_pointer_cast<UsdSceneItem>(item);
```

Comment:
I would do this differently.  If you get a nullptr argument, then returning a nullptr hierarchy interface is fine.

However, if you get a non-null argument, and the dynamic cast fails, that's an error.  You should
```
if (!item || !TF_VERIFY(usdItem)) {
   return nullptr;
} 
```